File: go/api/arch/member_test.go
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
 */

package arch

import (
	"api-demo/base"
	"api-demo/model"
	"testing"

	"github.com/stretchr/testify/assert"
)

const RootDeptNo = 1

var UserName = ""

func TestMemberApi(t *testing.T) {
	api := new(MemberApiClient)
	api.ApiClient = &base.ApiClient{
		ApiKey: base.ApiKey,
		Host:   base.Host,
	}

	t.Run("DeptMemberList", func(t *testing.T) {
		deptMemberList, err := api.DeptMemberList(RootDeptNo, map[string]interface{}{
			"hasChild": true,
		})
		if err != nil {
			t.Error(err)
		} else {
			t.Log(base.JsonIndentString(deptMemberList))
		}
	})

	t.Run("UserCreate", func(t *testing.T) {
		user, err := api.UserCreate("小云", map[string]interface{}{
			"username": "jiandaoyun",
		})
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "小云", user.User.Name)
			assert.Equal(t, "jiandaoyun", user.User.Username)
			t.Log(base.JsonIndentString(user))
			UserName = user.User.Username
		}
	})

	t.Run("UserInfo", func(t *testing.T) {
		user, err := api.UserInfo(UserName)
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "小云", user.User.Name)
			assert.Equal(t, "jiandaoyun", user.User.Username)
			t.Log(base.JsonIndentString(user))
		}
	})

	t.Run("UserUpdate", func(t *testing.T) {
		user, err := api.UserUpdate(UserName, map[string]interface{}{
			"name": "小简",
		})
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "小简", user.User.Name)
			t.Log(base.JsonIndentString(user))
		}
	})

	t.Run("UserDelete", func(t *testing.T) {
		response, err := api.UserDelete(UserName)
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "success", response.Status)
		}
	})

	t.Run("UserBatchDelete", func(t *testing.T) {
		response, err := api.UserBatchDelete([]string{UserName})
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "success", response.Status)
		}
	})

	t.Run("UserImport", func(t *testing.T) {
		response, err := api.UserImport([]model.Member{})
		if err != nil {
			t.Error(err)
		} else {
			assert.Equal(t, "success", response.Status)
		}
	})
}

File: python/test/api/arch/member_test.py
import unittest
from unittest.mock import patch
from src.model.user.user_create_param import UserCreateParam
from src.model.user.user_update_param import UserUpdateParam
from src.api.arch.member import MemberApiClient
from src.constants.http_constant import HttpConstant
import random

memberApiClient = MemberApiClient(HttpConstant.API_KEY, HttpConstant.HOST)

random = str(random.randint(100, 200))
name = 'name' + random
user_name = 'user_name' + random

deptNo = 1

user_create_param = UserCreateParam(name, user_name)
departments = [deptNo]
user_create_param.setDepartments(departments)

class TestSendEmail(unittest.TestCase):
    # 测试 创建成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_create(self, mock_send_post):
        memberApiClient.userCreate(user_create_param)
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/create")
        self.assertEqual(request_param.data.name, name)
        self.assertEqual(request_param.data.username, user_name)

    # 测试 获取部门成员（递归）
    @patch.object(memberApiClient, 'send_post')
    def test_dept_member_list(self, mock_send_post):
        memberApiClient.deptMemberList(deptNo, True)
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/department/user/list")
        self.assertEqual(request_param.data["dept_no"], deptNo)
        self.assertEqual(request_param.data["has_child"], True)

    # 测试 获取成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_info(self, mock_send_post):
        memberApiClient.userInfo(user_name)
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/get")
        self.assertEqual(request_param.data["username"], user_name)

    # 测试 更新成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_update(self, mock_send_post):
        user_update_param = UserUpdateParam(name, user_name)
        user_update_param.setDepartments([101, 201])
        user_update_param.setName(name + '_update')
        memberApiClient.userUpdate(user_update_param)
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/update")
        self.assertEqual(request_param.data["departments"], [101, 201])
        self.assertEqual(request_param.data["name"], name + '_update')
        self.assertEqual(request_param.data["username"], user_name)

    # 测试 删除成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_delete(self, mock_send_post):
        memberApiClient.userDelete(user_name)
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/delete")
        self.assertEqual(request_param.data["username"], user_name)

    # 测试 批量导入成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_import(self, mock_send_post):
        user_create_param.setUsername(user_name + '_import')
        user_create_param.setName(user_name + '_import')
        memberApiClient.userImport([user_create_param])
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/import")
        self.assertEqual(request_param.data["users"][0].username, user_name + '_import')
        self.assertEqual(request_param.data["users"][0].name, user_name + '_import')

    # 测试 批量删除成员
    @patch.object(memberApiClient, 'send_post')
    def test_user_batch_delete(self, mock_send_post):
        memberApiClient.userBatchDelete([user_create_param.username])
        request_param = mock_send_post.call_args[0][0]
        self.assertEqual(request_param.path, "/v5/corp/user/batch_delete")
        self.assertEqual(request_param.data["usernames"], [user_create_param.username])